File: java/spring-boot-rabbitmq/spring-boot-rabbitmq-publisher/src/main/java/com/meridal/examples/springbootamqp/service/RecordingService.java
package com.meridal.examples.springbootamqp.service;

import com.meridal.examples.springbootamqp.repository.RecordingRepository;
import com.meridal.examples.springbootrabbitmq.domain.Recording;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class RecordingService {

    private final RecordingRepository repository;

    public RecordingService(@Autowired RecordingRepository repository) {
        this.repository = repository;
    }

    public void publishRecording(final Recording recording) {
        this.repository.publish(recording);
    }
}

File: kotlin/kotlin-spring-boot-rabbitmq-publisher/src/main/kotlin/com/meridal/examples/kotlinspringbootamqp/service/RecordingService.kt
package com.meridal.examples.kotlinspringbootamqp.service

import com.meridal.examples.domain.Recording
import com.meridal.examples.kotlinspringbootamqp.repository.RecordingRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class RecordingService @Autowired constructor(private val repository: RecordingRepository) {

    fun publishRecording(recording: Recording) {
        repository.publish(recording)
    }
}